snuba/admin/audit_log/querylog.py

```diff
@@ -0,0 +1,66 @@
+from __future__ import annotations
+
+from datetime import datetime
+from enum import Enum
+from functools import partial
+from typing import Any, Callable, Mapping, MutableMapping, Union
+
+from snuba.clickhouse.native import ClickhouseResult
+
+DATETIME_FORMAT = "%B %d, %Y %H:%M:%S %p"
+
+from snuba.admin.audit_log.base import AuditLog
+
+
+class QueryExecutionStatus(Enum):
+    SUCCEEDED = "succeeded"
+    FAILED = "failed"
+
+
+class QuerylogAuditLog(AuditLog):
+    def _format_data(
+        self, user: str, timestamp: str, action: str, data: Mapping[str, Any]
+    ) -> Mapping[str, Any]:
+        status = data.get("status")
+        return {
+            "query": data.get("query"),
+            "status": status.value if status else "unknown",
+            "start_timestamp": timestamp,
+            "end_timestamp": datetime.now().strftime(DATETIME_FORMAT),
+        }
+
+
+__querylog_audit_log_notification_client = QuerylogAuditLog(__name__)
+
+
+def audit_log(
```

Comment:
What is the purpose of providing the alternative decorator interfaces here in addition to using the class directly? I don't think we have to provide multiple ways of doing the same thing here since we only really need to call this in one  place. This function is pretty heavily specific to the ClickHouseResult, and the other one is very specific to the runtime config scenario that it's hard to imagine it needs to be all that reusable.